apps/mobile/app/src/main/java/com/glycemicgpt/mobile/data/repository/AuthRepository.kt

```
@@ -168,4 +176,28 @@ class AuthRepository @Inject constructor(
     }
+
+    private suspend fun fetchSafetyLimits() {
+        try {
+            val response = api.getSafetyLimits()
+            if (response.isSuccessful) {
+                response.body()?.let { limits ->
+                    val min = limits.minGlucoseMgDl
+                    val max = limits.maxGlucoseMgDl
+                    val basal = limits.maxBasalRateMilliunits
+                    val bolus = limits.maxBolusDoseMilliunits
+                    if (min >= max || min < 20 || max > 500 || basal < 1 || basal > 15000 || bolus < 1 || bolus > 25000) {
```

detekt (1.23.8)

[warning] This condition is too complex (7). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)